src/Microsoft.ML.Core/Utilities/Utils.cs

```diff
@@ -615,37 +615,73 @@ public static string ExtractLettersAndNumbers(string value)
             return Regex.Replace(value, "[^A-Za-z0-9]", "");
         }
 
+        /// <summary>
+        /// Checks that an input IList is monotonically increasing.
+        /// </summary>
+        /// <param name="values">An array of values</param>
+        /// <returns>True if the array is monotonically increasing; false otherwise.
+        /// ILists containing NaN values are considered to be not monotonically increasing.</returns>
         public static bool IsSorted(IList<float> values)
         {
             if (Utils.Size(values) <= 1)
                 return true;
 
-            var prev = values[0];
-
-            for (int i = 1; i < values.Count; i++)
+            var previousValue = values[0];
+            var listLength = values.Count;
+            for (int i = 1; i < listLength; i++)
             {
-                if (!(values[i] >= prev))
+                var currentValue = values[i];
+                if (currentValue < previousValue)
```

Comment:
Switch check to match other implementations. `<` could be cheaper than `>=`, maybe.